src/main/java/org/zalando/fahrschein/PartitionManager.java

```diff
@@ -9,17 +9,19 @@
 
 public interface PartitionManager {
     /**
+     * Locks partitions for one event. When running consumers on multiple nodes they should try to lock partitions so that events are only processed by one consumer.
      *
      * @param eventName Name of the event
      * @param partitions The partitions to lock
      * @param lockedBy Name of one node of the consuming application
      * @param timeout Timeout after that the lock will automatically be released
      * @param timeoutUnit Time unit for the timeout
-     * @return Whether locking succeeded, i.e. the lock was not held by anyone or the existing lock expired. The current implementations will also return true if the lock is already held by the same node, but not update the expiry time.
+     * @return A {@code Lock} instance containing the list of locked partitions if locking succeded, {@code Optional.empty()} otherwise
```

Comment:
If you already started with Javadoc, why not `{@link org.zalando.fahrschein.domain.Lock}`?

(you have a type: "succeeded")